ofrak_core/ofrak/service/data_service.py

```diff
@@ -1,1464 +1,402 @@
-            if child_index != 0:
-                # The child_index corresponds to the first child that starts at or after the
-                # provided range. The child to the left must also be checked if it exists.
-                _, left_child = self._children[child_index - 1]
-                if target_range.overlaps(left_child.model.range):
-                    yield left_child
-
-            for i in range(child_index, len(self._children)):
-                # Iterate the children that starts at or after the provided range until the first
-                # child to the left of the range is found.
-                _, child = self._children[i]
-                if target_range.end < child.model.range.start:
-                    break
-                if target_range.overlaps(child.model.range) or target_range == child.model.range:
-                    # The ranges overlaps or they are equal (which is only possible if they are 0
-                    # sized ranges, which is considered an overlap)
-                    yield child
-        else:
-            self._children_tree = cast(IntervalTree, self._children_tree)
-            if target_range.length() > 0:
-                intervals = self._children_tree.overlap(target_range.start, target_range.end)
-            else:
-                intervals = self._children_tree.at(target_range.start)
-            for interval in sorted(intervals):
-                if interval.data.model.range.overlaps(target_range):
-                    yield interval.data
-            for i in range(child_index, len(self._children)):
-                # Handle the 0-sized children
-                _, child = self._children[i]
-                if child.model.range.start > target_range.end:
-                    break
-                if child.model.range.length() == 0 and (
-                    child.model.range.overlaps(target_range) or target_range == child.model.range
-                ):
-                    yield child
-
-    def get_unmapped_range(
-        self, position: int, relative_position: Optional[DataRangePosition] = None
-    ) -> Range:
-        if position < 0 or position > self.model.range.length():
-            raise OutOfBoundError(
-                f"The provided position {position} is outside the bounds of the parent {self}"
-            )
-        unmapped_range_index = cast(int, bisect.bisect_left(self._unmapped_ranges, (position,)))
-        if unmapped_range_index > 0:
-            unmapped_range_left: Optional[Range]
-            _, unmapped_range_left = self._unmapped_ranges[unmapped_range_index - 1]
-            if position < unmapped_range_left.end:
-                return unmapped_range_left
-            elif unmapped_range_left.end != position:
-                unmapped_range_left = None
-        else:
-            unmapped_range_left = None
+    async def create_root(self, data_id: DataId, data: bytes) -> DataModel:
```

Comment:
In my opinion having a type like `DataId` is more cognitive load that isn't worth it in the interface, but does make it easier to follow in the implementation. That's why I split it so that is only exists in the implementation, and that's where it's actually useful.

I think my comment at the definition of `DataId` is sufficient to explain that it means the same thing as `bytes`, I don't think it should be confusing.

Reply:
Ok